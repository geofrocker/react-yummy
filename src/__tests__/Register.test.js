import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Register from '../components/Register';

describe('register component', () => {
    const wrapper = shallow(<Register/>);
    const preventDefault = jest.fn();

    it('renders the Register class', () => {   
        expect(wrapper.find(".Register")).toHaveLength(1);
    });

    it('renders five div jsx elements', () => {
        expect(wrapper.find("div")).toHaveLength(8);        
    });
    it('renders the sign up form and submit data', () =>{
        wrapper.setState({name:'asiimwe', username:'elizei', email:'geofrocker2@gmail.com', passwordCandidate:"psalmsk", password:"psalms"});
        wrapper.find("#signup-form").simulate('submit', {preventDefault});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(preventDefault).toBeCalled();
    });

    it('render the two inputs', () =>{
        expect(wrapper.find("#password")).toHaveLength(1);
        expect(wrapper.find("#cpassword")).toHaveLength(1);
        expect(wrapper.find("#username")).toHaveLength(1);
        expect(wrapper.find("#name")).toHaveLength(1);
        expect(wrapper.find("#email")).toHaveLength(1);
        wrapper.find("#password").simulate('change', {target: {value: 'psalms'}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('make sure the email is valid', () =>{
      wrapper.find("#email").simulate('change', {target: {value: 'geofrocker@gmail.com'}});
      expect(toJson(wrapper)).toMatchSnapshot();
  })


    
})
