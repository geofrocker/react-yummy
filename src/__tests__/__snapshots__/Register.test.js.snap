// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register component make sure the email is valid 1`] = `
<div
  className="Register"
>
  <h1>
    Register Here
  </h1>
  <div
    className="col-xs-11 alert alert-danger"
  >
    Passwords do not match
  </div>
  <form
    id="signup-form"
    onSubmit={[Function]}
  >
    <div
      className="jumbotron col-xs-11"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value="asiimwe"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value="elizei"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value="geofrocker@gmail.com"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="cpassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value="psalmsk"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Comfirm Password"
          required={true}
          type="password"
          value="psalms"
        />
      </div>
      <input
        className="btn btn-primary pull-right"
        type="submit"
        value="Register"
      />
      <center>
        <p>
          or
        </p>
        <p>
          <a
            href="/login"
          >
            Already a member? Click here to login
          </a>
        </p>
      </center>
    </div>
  </form>
</div>
`;

exports[`register component render the two inputs 1`] = `
<div
  className="Register"
>
  <h1>
    Register Here
  </h1>
  <div
    className="col-xs-11 alert alert-danger"
  >
    Passwords do not match
  </div>
  <form
    id="signup-form"
    onSubmit={[Function]}
  >
    <div
      className="jumbotron col-xs-11"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value="asiimwe"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value="elizei"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value="geofrocker2@gmail.com"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="cpassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value="psalmsk"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Comfirm Password"
          required={true}
          type="password"
          value="psalms"
        />
      </div>
      <input
        className="btn btn-primary pull-right"
        type="submit"
        value="Register"
      />
      <center>
        <p>
          or
        </p>
        <p>
          <a
            href="/login"
          >
            Already a member? Click here to login
          </a>
        </p>
      </center>
    </div>
  </form>
</div>
`;

exports[`register component renders the sign up form and submit data 1`] = `
<div
  className="Register"
>
  <h1>
    Register Here
  </h1>
  <div
    className="col-xs-11 alert alert-danger"
  >
    Passwords do not match
  </div>
  <form
    id="signup-form"
    onSubmit={[Function]}
  >
    <div
      className="jumbotron col-xs-11"
    >
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value="asiimwe"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value="elizei"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value="geofrocker2@gmail.com"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="cpassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value="psalmsk"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Comfirm Password"
          required={true}
          type="password"
          value="psalms"
        />
      </div>
      <input
        className="btn btn-primary pull-right"
        type="submit"
        value="Register"
      />
      <center>
        <p>
          or
        </p>
        <p>
          <a
            href="/login"
          >
            Already a member? Click here to login
          </a>
        </p>
      </center>
    </div>
  </form>
</div>
`;
